### DISCUSSION TITLE
Result Gap between CTC and RNN-T Multi-Task Training

### DISCUSSION THREAD

**Discussion by teowenshen** [k2-fsa/icefall#973] [Q&A] [answered]
Between the non-streaming results of [vanilla](https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/RESULTS.md#pruned_transducer_stateless7-zipformer) and [CTC](https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/RESULTS.md#pruned_transducer_stateless7_ctc-zipformer-with-transducer-loss-and-ctc-loss), it is clear that the model with CTC performed better for test-other. However, is there any idea or intuition about why the models trained with a secondary CTC loss performed slightly worse for test-clean? 

<html><body>

  | test-clean | test-other | comment
-- | -- | -- | --
greedy search | 2.17 | 5.23 | --epoch 30 --avg 9 --max-duration 600
greedy search (with ctc) | 2.23 | 5.19 | --epoch 30 --avg 8
modified beam search | 2.15 | 5.20 | --epoch 30 --avg 9 --max-duration 600
modified beam search (with ctc) | 2.21 | 5.12 | --epoch 30 --avg 8
fast beam search | 2.15 | 5.22 | --epoch 30 --avg 9 --max-duration 600
fast beam search (with ctc) | 2.23 | 5.18 | --epoch 30 --avg 8

</body>
</html>

**Comment by yfyeung** [comment, marked as answer]
These gaps are so slight, I prefer to believe they are caused by noise.

**Reply by teowenshen** [reply]
Understood. Thanks for your reply!
